### PR TITLE
Update Davix builtin

### PR DESCRIPTION
**Pull request by amadio**

### CODE REVIEW

**Comment by amadio** [comment]
Duh, forgot to add the new `builtins/davix` directory to the commit...

**Comment by bellenot** [comment]
DAVIX is not supported and should be disabled on Windows.

**Comment by amadio** [comment]
Hi @bellenot thank you for letting me know. I will make sure it's disabled on Windows.

**Comment by amadio** [comment]
@phsft-bot build with flags -Dbuiltin_davix=ON

**Comment by amadio** [comment]
Checked locally. Failed on mac because it couldn't find OpenSSL. Should work with next command.

**Comment by amadio** [comment]
@phsft-bot build just on mac1013/native with flags -Dbuiltin_davix=ON -Dbuiltin_openssl=ON

File: builtins/davix/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  set(libname ${CMAKE_STATIC_LIBRARY_PREFIX}${lib}${CMAKE_STATIC_LIBRARY_SUFFIX})
+  list(APPEND DAVIX_LIBRARIES ${DAVIX_PREFIX}/lib/${libname})
+endforeach()
+
+ExternalProject_Add(DAVIX
+  URL ${DAVIX_URL}/${DAVIX_VERSION}/davix-embedded-${DAVIX_VERSION}.tar.gz
+  URL_HASH ${DAVIX_URLHASH}
+  PATCH_COMMAND patch -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/davix-0.6.4.patch
+  CMAKE_CACHE_ARGS -DCMAKE_PREFIX_PATH:STRING=${OPENSSL_PREFIX}
+  CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
```
Comment (Axel-Naumann) [inline builtins/davix/CMakeLists.txt]: Is `<INSTALL_DIR>` in `-DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>` a thing?

Comment (amadio) [reply builtins/davix/CMakeLists.txt]: Yes, it's the temporary installation directory where CMake's `ExternalProject_Add()` puts it.
From https://cmake.org/cmake/help/latest/module/ExternalProject.html:

> The command line, comment, working directory and byproducts of every standard and custom step are processed to replace the tokens <SOURCE_DIR>, <SOURCE_SUBDIR>, <BINARY_DIR>, <INSTALL_DIR> <TMP_DIR>, <DOWNLOAD_DIR> and <DOWNLOADED_FILE> with their corresponding property values defined in the original call to ExternalProject_Add().

**Comment by amadio** [comment]
Merged with OpenSSL update in #1845. Closing this PR.

**Pull request closed without merging**